src/platforms/java/common/source-context.mdx

```diff
@@ -5,3 +5,17 @@ description: "Learn about showing your source code as part of stack traces."
 ---
 
 <PlatformContent includePath="source-context" />
+
+## Setting Up Code Mappings
+
+In order to utilize features such as suspect commits and stack trace linking, you will need to set up a code mapping. This is a mapping between the source code in your repository and the source code in your stack traces. You can find information on how to set up code mappings in our [Set Up Code Mappings](/product/issues/suspect-commits/#2-set-up-code-mappings) docs.
+
+For Java projects, package names complicate the mapping. Before mapping the file name, Sentry uses custom logic to convert the frame data to a derived file path. Use the following steps to determine the correct **Stack Trace Root** **Source Code Root** for your code mapping:
+
+First, navigate to a stack trace that you wish to map. Find an In App frame, which is denoted by a bubble on the right side of the frame. The `module` will be shown as the first piece of text at the left hand side of the frame header (marked as **1** in the below screenshot). In this example, it is `com.example.vu.android.MainActivity`. Hovering over the module will show the absolute path to the file (marked as **2** in the below screenshot). In this example, it is `MainActivity.java`.
+
+![Highlighting where in the stack trace to find the module and absolute path](java-code-mappings-stacktrace.png)
+
+Next, calculate the derived file path. Take the frame `module` and replace all `.` characters with `/`. Then, replace the last item (`MainActivity` in this example) with the absolute path found earlier (`MainActivity.java`). In this example, the derived file path is `com/example/vu/android/MainActivity.java`.
+
+Finally, use the derived file path to determine the **Stack Trace Root** and **Source Code Root** using the steps detailed in the [Stack Trace Root and Source Code Root](/product/issues/suspect-commits/#stack-trace-root-and-source-code-root) docs.
```

Comment:
Instead of bailing back to the product docs, could we just include the instructions for setting stack trace root and source code root here? Afaict it wouldn't be that much extra text. We can still link back to the product docs as a "for more information..."

Reply:
Makes sense! Will add